### ISSUE TITLE
UnboundLocalError("local variable 'response' referenced before assignment")

### ISSUE THREAD

**Issue by Fak3** [encode/requests-async#55]
Version: 0.6.0
```python
import requests_async as requests
await requests.request(
            'POST', 'http://127.0.0.1:8000/api/v2/', timeout=10, data={}
        )
```

```
   File "/home/u1/.virtualenvs/paradox/lib/python3.7/site-packages/requests_async/api.py", line 6, in request
     return await session.request(method=method, url=url, **kwargs)
   File "/home/u1/.virtualenvs/paradox/lib/python3.7/site-packages/requests_async/sessions.py", line 79, in request
     resp = await self.send(prep, **send_kwargs)
   File "/home/u1/.virtualenvs/paradox/lib/python3.7/site-packages/requests_async/sessions.py", line 136, in send
     r = await adapter.send(request, **kwargs)
   File "/home/u1/.virtualenvs/paradox/lib/python3.7/site-packages/requests_async/adapters.py", line 65, in send
     await response.read()
ipdb> e
UnboundLocalError("local variable 'response' referenced before assignment")
ipdb> 
```

**Comment by moriW** [comment]
same error happend to me

**Comment by moriW** [comment]
same err happened to me

```python
Traceback (most recent call last):
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/requests_async/adapters.py", line 55, in send
    timeout=timeout,
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/http3/interfaces.py", line 49, in request
    return await self.send(request, verify=verify, cert=cert, timeout=timeout)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/http3/dispatch/connection_pool.py", line 121, in send
    raise exc
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/http3/dispatch/connection_pool.py", line 116, in send
    request, verify=verify, cert=cert, timeout=timeout
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/http3/dispatch/connection.py", line 56, in send
    response = await self.h2_connection.send(request, timeout=timeout)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/http3/dispatch/http2.py", line 40, in send
    await self.send_data(stream_id, data, timeout)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/http3/dispatch/http2.py", line 98, in send_data
    self.h2_state.send_data(stream_id, data)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/h2/connection.py", line 853, in send_data
    (frame_size, self.max_outbound_frame_size)
h2.exceptions.FrameTooLargeError: Cannot send frame size 35775, max frame size is 16384

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/mori/Developer/backendapp/bpbnapi/bpbn/misc/crontab/read_evaluate.py", line 63, in <module>
    asyncio.get_event_loop().run_until_complete(pull_evaluate_data(day))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 573, in run_until_complete
    return future.result()
  File "/Users/mori/Developer/backendapp/bpbnapi/bpbn/misc/crontab/read_evaluate.py", line 52, in pull_evaluate_data
    "keys": list(map(lambda x: f"p_{x.id}", paints))
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/requests_async/api.py", line 25, in post
    return await request("post", url, data=data, json=json, **kwargs)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/requests_async/api.py", line 6, in request
    return await session.request(method=method, url=url, **kwargs)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/requests_async/sessions.py", line 79, in request
    resp = await self.send(prep, **send_kwargs)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/requests_async/sessions.py", line 136, in send
    r = await adapter.send(request, **kwargs)
  File "/Users/mori/Developer/backendapp/bpbnapi/venv/lib/python3.7/site-packages/requests_async/adapters.py", line 65, in send
    await response.read()
UnboundLocalError: local variable 'response' referenced before assignment
```

**Comment by lovelydinosaur** [comment]
Resolved the `UnboundLocalError("local variable 'response' referenced before assignment")` and released as 0.6.1.

However @moriW's traceback at least is demonstrating an issue with the HTTP/2 implementation.

Could either of you try against 0.6.1, and include details on how to replicate the issue?

**Comment by lovelydinosaur** [comment]
Should all be resolved now with 0.6.2.

(Relevant http3 update... https://github.com/encode/http3/pull/98)
